syntax = "proto3";

service Database {
  rpc CreateUser(UserRegister) returns (UserResponse) {}
  rpc GetUser(User) returns (UserResponse) {}
  rpc GetUserByEmail(User) returns (UserResponse) {}
}

message UserRegister {
  string user_name = 1;
  string first_name = 2;
  string last_name = 3;
  string password = 4;
  string email = 5;
  string institution = 6;
  string country = 7;
}


message UserResponse {
  int32 status = 1;
  string msg = 2;
  repeated User data = 3;
}

message User {
  int32 id = 1;
  string user_name = 2;
  string first_name = 3;
  string last_name = 4;
  string email = 5;
  string institution = 6;
  string country = 7;
}
